packages/e2e-tests/src/specs/editor/fontCheck/fontCheck.js

```diff
@@ -0,0 +1,153 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  withExperimentalFeatures,
+  createNewStory,
+  visitAdminPage,
+  insertStoryTitle,
+  publishStory,
+  takeSnapshot,
+  addCustomFont,
+  removeCustomFont,
+} from '@web-stories-wp/e2e-test-utils';
+
+const createStoryWithTitle = async (title) => {
+  await createNewStory();
+  await insertStoryTitle(title);
+  await expect(page).toClick('#library-tab-text');
+  const insertButton = await page.waitForXPath(
+    `//button//span[contains(text(), 'Title 1')]`
+  );
+  await insertButton.click();
+  await expect(page).toMatchElement('[data-testid="textFrame"]', {
+    text: 'Title 1',
+  });
+};
+
+const updateFont = async (fontFamily) => {
+  await expect(page).toClick('button[aria-label="Font family"]');
+  await page.keyboard.type(fontFamily);
+  await page.keyboard.press('ArrowDown');
+  await page.keyboard.press('Enter');
+};
+
+const visitStoryWithTitle = async (title) => {
+  await visitAdminPage('edit.php', 'post_type=web-story');
+  await expect(page).toClick('a', { text: title });
+};
+
+const replaceFontWithFontPicker = async (fontFamily = '') => {
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toMatchElement('button', { text: 'Open anyway' });
+  await expect(page).toMatchElement('button', { text: 'Replace font' });
+  await takeSnapshot(page, 'Missing fonts dialog');
+  await expect(page).toClick(
+    "div[role='dialog'] div[class^='datalist__'] button"
+  );
+  await page.keyboard.type(fontFamily);
+  await page.keyboard.press('ArrowDown');
+  await page.keyboard.press('Enter');
+  await expect(page).toClick('button', { text: 'Replace font' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const replaceFontUsingDefault = async () => {
+  await page.waitForSelector('[role="dialog"]');
+  await expect(page).toClick('button', { text: 'Open anyway' });
+  await expect(page).toClick('[data-testid="textFrame"]');
+  await expect(page).toClick('button[aria-label="Font family"]');
+};
+
+const storyWithFontCheckDialog = async (
+  title,
+  customFont = 'Open Sans Condensed Light'
+) => {
+  // take steps needed to get Missing Font dialog to show
+  await addCustomFont();
+  await createStoryWithTitle(title);
+  await updateFont(customFont);
+  await publishStory();
+  await removeCustomFont();
+  await visitStoryWithTitle(title);
+  await page.waitForSelector('[data-testid="textFrame"]');
+  await page.waitForSelector('[role="dialog"]');
+};
+
+describe('Font Check', () => {
+  describe('Font Check Dialog', () => {
```

Comment:
Do we need nested description here? We only need it if we have multiple `describe`s. Let's remove this.

Reply:
Removed - https://github.com/GoogleForCreators/web-stories-wp/pull/10808/commits/c837a208eb93777368db9cd412ef5a4e99d1eb28